s3/src/main/scala/akka/stream/alpakka/s3/javadsl/S3Client.scala

```diff
@@ -481,11 +481,11 @@ final class S3Client(s3Settings: S3Settings, system: ActorSystem, mat: Materiali
               sse)
 
   private def toJava[M](
-      download: (akka.stream.scaladsl.Source[ByteString, M], Future[scaladsl.ObjectMetadata])
-  ): JPair[Source[ByteString, M], CompletionStage[ObjectMetadata]] = {
-    val (stream, meta) = download
-    JPair(stream.asJava, meta.map(metaDataToJava)(mat.executionContext).toJava)
-  }
+      download: Future[Option[(akka.stream.scaladsl.Source[ByteString, M], scaladsl.ObjectMetadata)]]
+  ): CompletionStage[Option[JPair[Source[ByteString, M], ObjectMetadata]]] =
```

Comment:
That's a fair bit of unwrapping. The `Option` would need conversion, as well, though.

Reply:
nice catch. Should be fixed now...